### PR TITLE
flac123: 0.0.12 -> 2.1.1, reformat, move to pkgs/by-name

### PR DESCRIPTION
**Pull request by kiike**
## Description of changes

This PR aims to upgrade the flac123 package, see if it [builds in Darwin](https://hydra.nixos.org/build/257557025#tabs-details) to help achieve ZHF #309482.

Upstream changelog:

> flac123-2.1.1
> 	* fix obscure remote mode bug evinced during testing
> 
> flac123-2.1.0
> 	* Remote mode stream destinations changed to stdout and stderr
> 	* --buffer-time option replaces --buffer-size passed to libao driver
> 
> flac123-2.0.0
> 	* Code refresh and maintenance
> 	* 24 bit depth support
> 	* 8 bit depth fix with proper wav output on macOS
> 	* pulse is Linux default
> 	* remove LEGACY_FLAC #define
> 	* update --help with modern driver names
> 	* move codebase to github
> 
> flac123-0.0.12
> 	* Extended remote spec with V(OLUME) command

## Things done

- Upgrade package
- Reformat using nixfmt-rfc-style
- Tried resulting binary on a directory with FLACs.
- Added myself to maintainers. I'm not an active user to this package but I guess it will need low effort to maintain. Tracking new releases via newreleases.io.

```
 /nix/store/7db1allqhvnpcd34151f41qgq0d77kkn-flac123-2.1.1/bin/flac123 /run/user/1000/mount/music/C*FLAC*/*.flac
flac123 version 2.1.1   'flac123 --help' for more info

Title  : ...
Album  : ...                               Year  : ...
Comment:  ...                              Genre : ...

Playing FLAC stream from 01. - ....flac
16 bit, 44100 Hz, 2 channels, 13933836 total samples, 315.96 total seconds
```

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fl/flac123/package.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}:
+
+stdenv.mkDerivation {
+  pname = "flac123";
+  version = "2.1.1";
+
+  src = fetchFromGitHub {
+    owner = "flac123";
+    repo = "flac123";
+    rev = "v2.1.1";
```
Comment (wegank) [inline pkgs/by-name/fl/flac123/package.nix:19]: I'd expect a string interpolation with `version` here instead of `v2.1.1`.

Comment (kiike) [reply pkgs/by-name/fl/flac123/package.nix:19]: Yep. You're right! Give me a minute to amend it.

**Comment by wegank** [comment]
Also, are there any breaking changes in this major version bump?

**Comment by kiike** [comment]
> Also, are there any breaking changes in this major version bump?

I'll check the release notes and get back to you.

File: pkgs/by-name/fl/flac123/package.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  stdenv,
+  fetchFromGitHub,
+  autoreconfHook,
+  flac,
+  libao,
+  libogg,
+  popt,
+}:
+
+stdenv.mkDerivation rec {
```
Comment (getchoo) [inline pkgs/by-name/fl/flac123/package.nix:12]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/fl/flac123/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  pname = "flac123";
+  version = "2.1.1";
+
+  src = fetchFromGitHub {
+    owner = "flac123";
+    repo = "flac123";
+    rev = "v${version}";
```
Comment (getchoo) [inline pkgs/by-name/fl/flac123/package.nix:19]:
```suggestion
    rev = "v${finalAttrs.version}";
```

File: pkgs/by-name/fl/flac123/package.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  nativeBuildInputs = [ autoreconfHook ];
+  buildInputs = [
+    flac
+    libao
+    libogg
+    popt
+  ];
+
+  meta = with lib; {
```
Comment (getchoo) [inline pkgs/by-name/fl/flac123/package.nix:31]:
```suggestion
  meta = {
```

File: pkgs/by-name/fl/flac123/package.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    homepage = "https://github.com/flac123/flac123";
+    description = "A command-line program for playing FLAC audio files";
+    license = licenses.gpl2Plus;
+    platforms = platforms.all;
```
Comment (getchoo) [inline pkgs/by-name/fl/flac123/package.nix:35]:
```suggestion
```
pretty sure this is the default. correct me if i'm wrong, though

File: pkgs/by-name/fl/flac123/package.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    flac
+    libao
+    libogg
+    popt
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/flac123/flac123";
+    description = "A command-line program for playing FLAC audio files";
+    license = licenses.gpl2Plus;
```
Comment (getchoo) [inline pkgs/by-name/fl/flac123/package.nix:34]:
```suggestion
    license = lib.licenses.gpl2Plus;
```

File: pkgs/by-name/fl/flac123/package.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    platforms = platforms.all;
+    maintainers = with maintainers; [ kiike ];
```
Comment (getchoo) [inline pkgs/by-name/fl/flac123/package.nix:36]:
```suggestion
    maintainers = with lib.maintainers; [ kiike ];
```

File: pkgs/by-name/fl/flac123/package.nix [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    mainProgram = "flac123";
+  };
+}
```
Comment (getchoo) [inline pkgs/by-name/fl/flac123/package.nix:39]:
```suggestion
})
```

**Review by getchoo — CHANGES_REQUESTED** [review]

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 310150` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>flac123</li>
  </ul>
</details>

**Comment by kiike** [comment]
Thanks for the help @wegank!

**Pull request merged**